### PR TITLE
Attempt to integrate defmain for command line args

### PR DESCRIPTION
**Pull request by abishek**
I am trying to add command line arguments to pass into the process function. Please let me know if this is in the right direction. If not, do let me know how to fix it so I can learn to use defmain package.

### CODE REVIEW

File: nastafile.lisp [hunk 1]
```diff
@@ -482,3 +482,11 @@ As the second value it returns ids of root articles.
   (dump-trees *all-articles*)
   (add-nojekyll-file))
 
+
+(defmain:defmain main
```
Comment (svetlyak40wt) [inline nastafile.lisp:486]: Export the `main` function from the package.

File: nastafile.asd [hunk 2]
```diff
@@ -18,6 +18,7 @@
                "group-by"
                "rutils"
                "log4cl-extras"
-               "trivial-garbage")
+               "trivial-garbage"
+               "defmain")
```
Comment (svetlyak40wt) [inline nastafile.asd:22]: Let's add a Deploy here to build a binary using `(asdf:make :nastafile)`.

Here is how to do this: https://github.com/Shinmera/deploy#how-to

File: nastafile.lisp (hunk 1 above)
Comment (abishek) [reply nastafile.lisp:486]: Cool. Is the usage correct there? I'll export the function and resubmit the PR.

Comment (svetlyak40wt) [reply nastafile.lisp:486]: Just push new commit

**Comment by abishek** [comment]
Updated. Please check and confirm. Also, I am not able to test in my machine. Something incorrectly setup. Will fix soon :)

**Review by svetlyak40wt — APPROVED** [review]
Great!

Now try to run:

```
qlot exec ros --eval '(asdf:make :nastafile)' --eval '(uiop:quit)'
```

**Comment by abishek** [comment]
> Great!
> 
> Now try to run:
> 
> ```
> qlot exec ros --eval '(asdf:make :nastafile)' --eval '(uiop:quit)'
> ```

It runs but doesn't create a binary in the bin folder. No errors either.

**Comment by abishek** [comment]
> Great!
> 
> Now try to run:
> 
> ```
> qlot exec ros --eval '(asdf:make :nastafile)' --eval '(uiop:quit)'
> ```

I get this error:

```
((FLET DEPLOY::FOREIGN-LIBRARIES :IN "/Users/abishek/Projects/nastafile/.qlot/dists/deploy/software/deploy-master/deploy.lisp") :DIRECTORY #P"/Users/abishek/Projects/nastafile/bin/")
   source: (RESTART-CASE (ERROR
                          "~a does not have a known shared library file path."
                          LIB)
             (PROVIDE-PATH (PATH) :REPORT
              "Provide the path to the library manually." :INTERACTIVE
              QUERY-FOR-LIBRARY-PATH (SETF (LIBRARY-PATH LIB) PATH)))
```

**Pull request merged**